clusterloader2/testing/dra-extended-resources/README.md

```diff
@@ -0,0 +1,183 @@
+# DRA Extended Resources Scale Test
+
+## Overview
+
+This test validates the performance and scalability of Kubernetes' DRA Extended Resources feature (KEP-5004). It measures how well the scheduler handles extended resource requests that are backed by Dynamic Resource Allocation (DRA), allowing applications to use familiar extended resource syntax while benefiting from DRA's dynamic allocation capabilities.
+
+## What This Test Does
+
+This test scenario mirrors the structure of the regular DRA test (`testing/dra/config.yaml`) but uses **extended resources syntax** instead of explicit ResourceClaims:
+
+1. **Setup Phase**: Creates a DeviceClass with an `extendedResourceName` field that maps DRA devices to traditional extended resources
+2. **Fill Phase**: Fills the cluster to 90% utilization with long-running pods that request extended resources (e.g., `example.com/gpu: 1`)
+3. **Measurement Phase**: Measures performance while continuously scheduling short-lived pods at a steady rate using the same extended resource requests
+4. **Metrics Collection**: Collects detailed metrics on:
+   - Pod startup latency
+   - Job lifecycle latency  
+   - Scheduler performance metrics
+   - DRA-specific metrics (PrepareResources/UnprepareResources latencies)
+   - Extended resource allocation metrics
+
+## Key Differences from Regular DRA Test
+
+- **No ResourceClaimTemplates**: Uses DeviceClass with `extendedResourceName` instead
+- **Extended Resource Syntax**: Pods request `example.com/gpu: 1` in `resources.limits` instead of using `resourceClaims`
+- **Transparent DRA**: The scheduler automatically creates ResourceClaims behind the scenes
+- **Backward Compatibility**: Tests that existing extended resource workloads work with DRA
+
+## Prerequisites
+
+1. **Feature Gate**: Ensure `DRAExtendedResource=true` is enabled on:
+   - kube-apiserver
+   - kube-scheduler  
+   - kubelet
+
+2. **DRA Driver**: A DRA driver must be running (installed automatically by the test)
+
+3. **Prometheus**: Required for metric-based measurements
+
+## Usage
+
+### Environment Variables
+
+```bash
+export CL2_MODE=Indexed                    # Job completion mode
+export CL2_NODES_PER_NAMESPACE=1           # Namespaces per node
+export CL2_LOAD_TEST_THROUGHPUT=20         # Fast initial fill rate
+export CL2_STEADY_STATE_QPS=5              # Controlled rate for measurement
+export CL2_JOB_RUNNING_TIME=30s            # Short-lived pods runtime
+export CL2_LONG_JOB_RUNNING_TIME=1h        # Long-running pods runtime
+export CL2_GPUS_PER_NODE=8                 # Extended resources per node
+export CL2_FILL_PERCENTAGE=90              # Cluster fill percentage
+export CL2_EXTENDED_RESOURCE_NAME="example.com/gpu"  # Extended resource name
+```
+
+### Run the Test
+
+```bash
+# Make sure a Prometheus stack is deployed
+./run-e2e.sh cluster-loader2 \
+--provider=kind \
```

Comment:
Based on my experience with load scenario default kind cluster configuration doesn't not fulfill some prerequisites (e.g. metric endpoints of KCM/scheduler are not exposed). Have you verified that it works?